packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -96,14 +96,14 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
         }));
         this.connectionProvider.listen({
             path: LanguageContribution.getPath(this),
-            onConnection: messageConnection => {
+            onConnection: async messageConnection => {
                 if (toDeactivate.disposed) {
                     messageConnection.dispose();
                     return;
                 }
                 toDeactivate.push(messageConnection);
 
-                const languageClient = this.createLanguageClient(messageConnection);
+                const languageClient = await this.createLanguageClient(messageConnection);
```

Comment:
if a connection is disposed in the meantime then we create a language client for a broken connection. I don't think this code can be async. Would it be possible to prefetch java extension as part of `waitForActivation` only once?